packages/web/src/pages/collection-page/CollectionPageProvider.tsx

```diff
@@ -340,11 +345,19 @@ class CollectionPage extends Component<
 
   fetchCollection = (pathname: string, forceFetch = false) => {
     const params = parseCollectionRoute(pathname)
-    if (params) {
-      const { collectionId } = params
-      if (forceFetch || collectionId !== this.state.playlistId) {
-        this.setState({ playlistId: collectionId as number })
-        this.props.fetchCollection(collectionId as number)
+
+    if (params?.collectionId) {
+      if (forceFetch || params?.collectionId !== this.state.playlistId) {
```

Comment:
ultra nit, params.collectionId since if branch ensures it's there few other cases down below too. personally, once ive established existance in an object, and its reffed multple times, i desturcture it,
```
if (params.?collectionId) {
  const {collectionId} = params
  ...
```